File: app/src/main/java/com/mytlogos/enterprise/background/api/ListMediaApi.java
package com.mytlogos.enterprise.background.api;

import com.mytlogos.enterprise.background.api.model.ClientMedium;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

interface ListMediaApi {

    @GET
    Call<List<ClientMedium>> getListMedia(@Url String url, @QueryMap Map<String, Object> body);

    @POST
    Call<Boolean> addListMedia(@Url String url, @Body Map<String, Object> body);

    @DELETE
    Call<Boolean> deleteListMedia(@Url String url, @Body Map<String, Object> body);

    @PUT
    Call<Boolean> updateListMedia(@Url String url, @Body Map<String, Object> body);
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/ListMediaApi.kt
package com.mytlogos.enterprise.background.api

import com.mytlogos.enterprise.background.api.model.ClientListQuery
import retrofit2.Response
import retrofit2.http.*

internal interface ListMediaApi {
    @GET
    suspend fun  getListMedia(@Url url: String, @QueryMap body: MutableMap<String, Any?>): Response<ClientListQuery>

    @POST
    suspend fun addListMedia(@Url url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>

    @HTTP(method = "DELETE", hasBody = true)
    suspend fun deleteListMedia(@Url url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>

    @PUT
    suspend fun updateListMedia(@Url url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>
}